lib/archethic/transaction_chain/transaction/proof_of_validation.ex

```diff
@@ -0,0 +1,178 @@
+    nb_required_stamps = get_nb_required_validations(sorted_nodes)
+
+    if nb_valid_stamps >= nb_required_stamps do
+      :reached
+    else
+      nb_validation_nodes = get_nb_validation_nodes(sorted_nodes)
+      nb_remaining_stamps = nb_validation_nodes - Enum.count(stamps)
+
+      # If the remaining stamp to receive cannot reach the required validations we return an error
+      if nb_valid_stamps + nb_remaining_stamps >= nb_required_stamps,
+        do: :not_reached,
+        else: :error
+    end
+  end
+
+  @doc """
+  Construct the proof of validation aggregating valid cross stamps
+  """
+  @spec create(nodes :: SortedNode.t(), stamps :: list({Crypto.key(), CrossValidationStamp.t()})) ::
+          t()
+  def create(%SortedNode{nodes: nodes}, stamps) do
+    valid_cross = filter_valid_cross_stamps(stamps)
+
+    {public_keys, signatures} =
+      Enum.reduce(
+        valid_cross,
+        {[], []},
+        fn {_from, %CrossValidationStamp{node_public_key: public_key, signature: signature}},
+           {public_keys, signatures} ->
+          {[public_key | public_keys], [signature | signatures]}
+        end
+      )
+
+    aggregated_signature = Crypto.aggregate_signatures(signatures, public_keys)
+
+    bitmask =
+      Enum.reduce(valid_cross, <<>>, fn {from, _}, acc ->
+        index = Enum.find_index(nodes, &(&1.first_public_key == from))
+        Utils.set_bitstring_bit(acc, index)
+      end)
+
+    %__MODULE__{signature: aggregated_signature, nodes_bitmask: bitmask}
+  end
+
+  @doc """
+  Returns the list of node that signed the proof of validation
+  """
+  @spec get_nodes(nodes :: SortedNode.t(), proof :: t()) :: list(Node.t())
+  def get_nodes(%SortedNode{nodes: nodes}, %__MODULE__{nodes_bitmask: bitmask}) do
+    bitmask
+    |> Utils.bitstring_to_integer_list()
+    |> Enum.with_index()
+    |> Enum.filter(&(elem(&1, 0) == 1))
+    |> Enum.map(fn {_, index} -> Enum.at(nodes, index) end)
+  end
+
+  @doc """
+  Validate a proof of validation
+  - Number of validation reach the threshold
+  - aggregated signature is valid
+  """
+  @spec valid?(nodes :: SortedNode.t(), proof :: t(), validation_stamp :: ValidationStamp.t()) ::
+          boolean()
+  def valid?(
+        sorted_nodes = %SortedNode{},
+        proof = %__MODULE__{signature: signature},
+        validation_stamp
+      ) do
+    nb_required_stamps = get_nb_required_validations(sorted_nodes)
+    validation_nodes = get_nodes(sorted_nodes, proof)
+
+    if Enum.count(validation_nodes) >= nb_required_stamps do
+      aggregated_public_key =
+        validation_nodes
+        |> Enum.map(& &1.mining_public_key)
+        |> Crypto.aggregate_mining_public_keys()
+
+      raw_data = CrossValidationStamp.get_row_data_to_sign(validation_stamp, [])
+
+      Crypto.verify?(signature, raw_data, aggregated_public_key)
+    else
+      false
+    end
+  end
+
+  defp filter_valid_cross_stamps(stamps) do
+    {valid_cross, _invalid_cross} =
```

Comment:
Suggestion: Enum.filter

Reply:
Question: What is the purpose of using `split_with`?